restli-server/src/main/java/com/linkedin/restli/internal/server/response/ResponseUtils.java

```diff
@@ -84,6 +91,90 @@ static Object translateCanonicalKeyToAlternativeKeyIfNeeded(Object canonicalKey,
     }
   }
 
+  /**
+   *
+   * @param dataSchema
+   * @param dataMap
+   * @return a new different data map that contains original data plus the default values
+   */
+  public static DataMap fillInDefaultValues(DataSchema dataSchema, DataMap dataMap)
+  {
+    DataMap dataDefaultFilled = new DataMap();
+    if (dataMap != null)
+    {
+      dataDefaultFilled.putAll(dataMap);
```

Comment:
Why do you need to copy?

Reply:
It is for return, If I don't return a `DataMap`, then for non existing field that has a default value, I will need to pass in a `DataMap` for putting data.